pkg/healthcheck/healthcheck.go

```diff
@@ -2120,7 +2121,7 @@ func (hc *HealthChecker) checkProxyInitRunsAsRoot(ctx context.Context, config *l
 		}
 	}
 	if hasDockerNodes && !runAsRoot {
```

Comment:
if we get to this point, we already know that runAsRoot must be false since we didn't exit early on line 2102.

I think we should either keep this the way it was before where we pass in Helm values and validate the cluster state against those *or* if we're going to change this so that the caller is responsible for extracting the runAsRoot value, then the caller should also be responsible for deciding if it needs to call this check based on that value.  In the latter case, this function could be called `CheckNodesHaveNonDockerRuntime` or something like that.

Reply:
I like the latter suggestion. I changed it to `CheckNodesHaveNonDockerRuntime` and we only check that when proxyInit is _not_ running as root.